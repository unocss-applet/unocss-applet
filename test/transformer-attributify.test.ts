import { promises as fs } from 'node:fs'
import path from 'node:path'
import { describe, expect, it } from 'vitest'
import type { SourceCodeTransformer } from '@unocss/core'
import { createGenerator } from '@unocss/core'
import MagicString from 'magic-string'
import transformerAttributify from '@unocss-applet/transformer-attributify'
import presetUno from '@unocss/preset-uno'
import { presetAttributify, presetIcons } from 'unocss'
import { presetExtra } from 'unocss-preset-extra'

async function transform(code: string, transformer: SourceCodeTransformer) {
  const s = new MagicString(code)

  const uno = createGenerator({
    presets: [
      presetUno(),
      presetIcons({
        scale: 1.2,
        cdn: 'https://esm.sh/',
      }),
      presetAttributify(),
      presetExtra(),
    ],
    transformers: [
      transformer,
    ],
  })
  await transformer.transform(s, 'foo.vue', { uno, tokens: new Set() } as any)
  return s.toString()
}

describe('transformer-attributify', async () => {
  const content = await fs.readFile(
    path.resolve(__dirname, './fixtures/attributify.vue'),
  )

  it('basic', async () => {
    const transformer = transformerAttributify({ ignoreAttributes: ['block'] })
    const result = await transform(content.toString(), transformer)
    expect(result).toMatchSnapshot()
  })

  it('prefixedOnly', async () => {
    const transformer = transformerAttributify({ prefixedOnly: true })
    const result = await transform(content.toString(), transformer)
    expect(result).toMatchSnapshot()
  })
})
