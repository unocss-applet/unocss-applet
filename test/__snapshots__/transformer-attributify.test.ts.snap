// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`transformer-attributify > basic 1`] = `
"<script setup lang="ts">
import { ref } from 'vue'

const bool = ref<boolean>()
const c = 'text-red'
</script>

<template>
  <div class="text-center aaa p-2.5 text-blue" :class="c">
    <div  class="text-4xl ma text-green p-2"/>
    <div class="bg-pink dark:bg-yellow hover:bg-green">
      dark
    </div>
    <div :class="\`p-2.5 \${bool ? 'p-0.5 m-0.5' : ''}\`" :hover-class="['!bg-green']"  class="m-2"/>
    <div class="p-1 flex flex-col flex-gap-1 items-center text-#fff" />
    <div class="b b-green text-right h-10 !text-pink">
      123456789
    </div>
    <div class="text-purple b b-green" :hover-class="['!bg-green', bool ? '' : 'text-red']">
      123456789
    </div>
    <div  class="i-carbon:logo-twitter dark:i-carbon:logo-github"/>
    <div  class="i-carbon-logo-twitter dark:i-carbon-logo-github mt-2.5"/>
    <div class="[&>div]:w-10 text-red">
      <div block class="text-left">
        1
      </div>
      <div class="color-[#233] text-center">
        2
      </div>
      <div class="text-right">
        3
      </div>
    </div>
    <div class="size-10 size-20" />
  </div>
</template>
"
`;

exports[`transformer-attributify > prefixedOnly 1`] = `
"<script setup lang="ts">
import { ref } from 'vue'

const bool = ref<boolean>()
const c = 'text-red'
</script>

<template>
  <div class="text-center aaa" p="2.5" text-blue :class="c">
    <div text="4xl" ma text-green  class="p-2"/>
    <div bg="pink dark:yellow hover:green">
      dark
    </div>
    <div :class="\`p-2.5 \${bool ? 'p-0.5 m-0.5' : ''}\`" m-2 :hover-class="['!bg-green']" />
    <div flex="~ col gap-1" class="p-1" items-center text="#fff" />
    <div text-right h-10 text="!pink" class="b b-green">
      123456789
    </div>
    <div class="text-purple b b-green" :hover-class="['!bg-green', bool ? '' : 'text-red']">
      123456789
    </div>
    <div i-carbon:logo-twitter dark:i-carbon:logo-github />
    <div i-carbon-logo-twitter dark:i-carbon-logo-github mt-2.5 />
    <div class="[&>div]:w-10" text-red>
      <div block text-left>
        1
      </div>
      <div text-center class="color-[#233]">
        2
      </div>
      <div text-right>
        3
      </div>
    </div>
    <div class="size-10" size="20" />
  </div>
</template>
"
`;
